src/include/network/lan/address.rb

```diff
@@ -1117,82 +1073,87 @@ def AddressDialog
+        # LanItems.add_current_device_to_routing if LanItems.update_routing_devices?
       end
 
-      if LanItems.type == "vlan"
-        LanItems.vlan_etherdevice = Ops.get_string(@settings, "ETHERDEVICE", "")
-        LanItems.vlan_id = Builtins.tostring(
-          Ops.get_integer(@settings, "VLAN_ID", 0)
+      if builder.type == "vlan"
+        builder.set(option: "ETHERDEVICE", value: Ops.get_string(@settings, "ETHERDEVICE", ""))
+        builder.set(
+          options: "VLANID",
+          value:   Builtins.tostring(Ops.get_integer(@settings, "VLAN_ID", 0))
         )
       elsif LanItems.type == "br"
-        LanItems.bridge_ports = @settings["BRIDGE_PORTS"].join(" ")
-        log.info "bridge ports stored as #{LanItems.bridge_ports.inspect}"
+        builder.set(option: "BRIDGE_PORTS", value: @settings["BRIDGE_PORTS"].join(" "))
+        log.info "bridge ports stored as #{builder.option("BRIDGE_PORTS")}"
       elsif LanItems.type == "bond"
-        new_slaves = @settings.fetch("SLAVES", []).select { |s| !LanItems.bond_slaves.include? s }
-        LanItems.bond_slaves = @settings["SLAVES"]
-        LanItems.bond_option = @settings["BONDOPTION"]
+        new_slaves = @settings.fetch("SLAVES", []).select do |s|
+          # TODO: check initialization of "SLAVES"
+          !builder.option("SLAVES").include? s
+        end
+        builder.set(option: "SLAVES", value: @settings["SLAVES"])
+        builder.set(option: "BONDOPTION", value: @settings["BONDOPTION"])
         Lan.autoconf_slaves = (Lan.autoconf_slaves + new_slaves).uniq.sort
-        log.info "bond settings #{LanItems.bond_slaves}"
-      elsif Builtins.contains(["tun", "tap"], LanItems.type)
-        LanItems.tunnel_set_owner = Ops.get_string(
-          @settings,
-          "TUNNEL_SET_OWNER",
-          ""
+        log.info "bond settings #{builder.option("BONDOPTION")}"
+      elsif Builtins.contains(["tun", "tap"], builder.type)
+        builder.set(
+          option: "TUNNEL_SET_OWNER",
+          value: Ops.get_string(@settings, "TUNNEL_SET_OWNER", "")
         )
-        LanItems.tunnel_set_group = Ops.get_string(
-          @settings,
-          "TUNNEL_SET_GROUP",
-          ""
+        builder.set(
+          option: "TUNNEL_SET_GROUP",
+          value:  Ops.get_string(@settings, "TUNNEL_SET_GROUP", "")
         )
       end
 
       # proceed with WLAN settings if appropriate, #42420
-      ret = :wire if ret == :next && LanItems.type == "wlan"
+      ret = :wire if ret == :next && builder.type == "wlan"
 
-      deep_copy(ret)
+      ret
     end
 
   private
 
     # Initializes the Address Dialog @settings with the corresponding LanItems values
-    def initialize_address_settings
-      @settings.replace( # general tab:
-        "STARTMODE"        => LanItems.startmode,
-        "IFPLUGD_PRIORITY" => LanItems.ifplugd_priority,
+    def initialize_address_settings(builder)
+      @settings.replace(
```

Comment:
so in future we will pass builder instead of @settings to CWM widgets, right?

Reply:
I'd like to. Original idea was to keep "open" interface in it and either submit or drop it as the user wishes (confirms Add / Edit or cancel it)